js/src/admin/utils/getCategorizedExtensions.js

```diff
@@ -0,0 +1,29 @@
+export default function getCategorizedExtensions() {
+  let extensions = {};
+
+  Object.keys(app.data.extensions).map((id) => {
+    const extension = app.data.extensions[id];
+    let category = extension.extra['flarum-extension'].category;
+
+    if (!extension.extra['flarum-extension'].category) {
+      category = 'other';
+    }
+
+    // Wrap languages packs into new system
+    if (extension.extra['flarum-locale']) {
+      category = 'language';
+    }
+
+    if (category in app.extensionCategories) {
+      extensions[category] = extensions[category] || {};
```

Comment:
Why implement the list of extensions for each category as an object instead of an array?